src/xml_comment.cc

```diff
@@ -28,17 +28,13 @@ NAN_METHOD(XmlComment::New) {
   assert(document);
 
   v8::Local<v8::Value> contentOpt;
-  if(info[1]->IsString()) {
+  if (info[1]->IsString()) {
       contentOpt = info[1];
   }
   v8::String::Utf8Value contentRaw(contentOpt);
   const char* content = (contentRaw.length()) ? *contentRaw : NULL;
 
-  xmlChar* encoded = content ? xmlEncodeSpecialChars(document->xml_obj, (const xmlChar*)content) : NULL;
-  xmlNode* comm = xmlNewDocComment(document->xml_obj,
-                                   encoded);
-  if (encoded)
```

Comment:
From my understanding encoded in was pointing to newly allocated memory so content does not need to be explicitly freed